src/learning/ProfileInitializer.cpp

```diff
@@ -199,16 +201,25 @@ std::vector<Perf> ProfileInitializer::initializeProfilePerformance(
 }
 
 void ProfileInitializer::initializeProfiles(MRSortModel &model) {
-  std::vector<Performance> perf_vec;
-  Performance firstAltPerf = altPerformance_.getPerformanceTable()[0];
-  std::vector<std::string> criteriaIds = firstAltPerf.getCriterionIds();
-  std::vector<float> catFreq = ProfileInitializer::categoryFrequency();
-  for (std::string criterion : criteriaIds) {
-    // OPTIM : POSSIBILITY parallelization asynchrone
-    std::vector<Perf> p = ProfileInitializer::initializeProfilePerformance(
-        criterion, model.categories, catFreq);
-    perf_vec.push_back(Performance(p));
+  bool ordered = 0;
+  while (!ordered) {
```

Comment:
Maybe, should we add a maximum iteration in the loop? Just to make sure we crash and return an error telling what's going on instead of just getting stuck in an infinite loop (if it ever happens)

Reply:
I added an if statement